docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_global_services_logging_duplicated.mdx

```diff
@@ -23,6 +23,299 @@ HIPAA
+
+   ```
+   aws cloudtrail lookup-events --lookup-attributes AttributeKey=EventName,AttributeValue=eventName
+   ```
+   Replace 'eventName' with the name of the event you want to check for duplicates. This command will return a list of all the events with the specified name.
+
+4. To detect duplicate entries, you can write a Python script that parses the JSON output from the above command and checks for duplicate entries. Here is a simple example:
+
+   ```python
+   import json
+   import subprocess
+
+   def get_cloudtrail_events(event_name):
+       command = ['aws', 'cloudtrail', 'lookup-events', '--lookup-attributes', 'AttributeKey=EventName,AttributeValue={}'.format(event_name)]
+       process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+       stdout, stderr = process.communicate()
+       if process.returncode != 0:
+           raise Exception('Error executing command: {}'.format(stderr.decode('utf-8')))
+       return json.loads(stdout.decode('utf-8'))
+
+   def check_for_duplicates(events):
+       event_ids = [event['EventId'] for event in events['Events']]
+       if len(event_ids) != len(set(event_ids)):
+           print('Duplicate entries found')
+       else:
+           print('No duplicate entries found')
+
+   events = get_cloudtrail_events('eventName')
+   check_for_duplicates(events)
+   ```
+   Replace 'eventName' with the name of the event you want to check for duplicates. This script will print 'Duplicate entries found' if there are duplicate entries in the CloudTrail logs, and 'No duplicate entries found' otherwise.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Import necessary libraries and establish a session**: First, you need to import the necessary libraries in your Python script. You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. After importing the libraries, establish a session with AWS using your access key and secret access key.
+
+    ```python
+    import boto3
+
+    session = boto3.Session(
+        aws_access_key_id='YOUR_ACCESS_KEY',
+        aws_secret_access_key='YOUR_SECRET_KEY',
+        region_name='YOUR_REGION'
+    )
+    ```
+
+2. **Create CloudTrail client**: After establishing a session, create a CloudTrail client. This client will allow you to interact with the CloudTrail service and fetch the necessary logs.
+
+    ```python
+    cloudtrail = session.client('cloudtrail')
+    ```
+
+3. **Fetch and analyze CloudTrail logs**: Now, fetch the CloudTrail logs using the lookup_events() function. This function returns a dictionary containing the CloudTrail event list. You can then iterate over this list and check for duplicate entries.
+
+    ```python
+    response = cloudtrail.lookup_events(
+        LookupAttributes=[
+            {
+                'AttributeKey': 'EventName',
+                'AttributeValue': 'CreateTrail'
+            },
+        ],
+    )
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>